src/compiler/program.ts

```diff
@@ -176,6 +176,39 @@ namespace ts {
         return undefined;
     }
 
+   function tryReadMainSection(packageJsonPath: string, baseDirectory: string, state: ModuleResolutionState): string {

```

Comment:
Why do we go to such great lengths if we end up just calling `require()` in `sys.loadExtension`?